src/theory/arith/constraint.h

```diff
@@ -651,6 +651,13 @@ class Constraint {
   /**
    * @brief Returns whether this constraint is provable using a Farkas
    * proof that has input assertions as its antecedents.
+   *
+   * An example of a constraint that has a simple Farkas proof:
+   *    x <= 0 proven from x + y <= 0 and x - y <= 0.
+   *
+   * An example of a constraint that does not have a simple Farkas proof:
+   *    x <= 0 proven from x + y <= 0 and x - y <= 0.5 for integers x, y
+   *       (since integer reasoning is also required!).
```

Comment:
I think integer QF_LIRA is further down the road - is there a simple example of a purely real linear constraint that we currently cannot prove?

Reply:
Other than integer reasoning, there is "trichotomy" reasoning -- the fact that exactly one of (x<a), (x=a), (x>a) can be true.

[This method](https://github.com/CVC4/CVC4/blob/master/src/theory/arith/constraint.cpp#L1156) is used to record use of this kind of reasoning. Unfortunately, I have not been able to find a test case that actually uses this kind of reasoning (it seems to overlap with preprocessing/boolean  reasoning/farkas reasoning?). 

Anyways, I've added a hypothetical use of trichotomy to the documentation, just not an actual use. If I find a test case that actually triggers it, I'll add that test case to the documentation.